### PR TITLE
Add contributing docs specific to rule-testing patterns

### PR DESCRIPTION
**Pull request by jlaneve**
## Summary

This PR adds a section to the contributing guide addressing Ruff's patterns for testing rules! I wrote my first rule a few hours ago and hadn't worked with this kind of testing before, so I figured it'd be good to explicitly addressing this.

## Test Plan

n/a

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -154,6 +145,24 @@ This implies that rule names:
 When re-implementing rules from other linters, this convention is given more importance than
 preserving the original rule name.
 
+#### Rule testing: fixtures and snapshots
+
+To test rules, Ruff uses snapshots of Ruff's output for a given file (fixture). Generally, there will be one file per rule (e.g., `E402.py`), and each file will contain all necessary examples of both violations and non-violations. `cargo insta review` will generate a snapshot file containing Ruff's output for each fixture, which you can then commit alongside your changes.
+
+Once you've completed the code for the rule itself, you can define tests with the following steps:
+
+1. Add a Python file to `crates/ruff/resources/test/fixtures/[linter]` that contains the code you want to test. The file name should match the rule name (e.g., `E402.py`), and it should include examples of both violations and non-violations.
```
Comment (JonathanPlasse) [inline CONTRIBUTING.md:154]: If you use the script `add_rule.py` the file is created automatically.

**Comment by charliermarsh** [comment]
These are great. Thanks tons, @jlaneve!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -154,6 +153,36 @@ This implies that rule names:
 When re-implementing rules from other linters, this convention is given more importance than
 preserving the original rule name.
 
+#### Rule testing: fixtures and snapshots
```
Comment (charliermarsh) [inline CONTRIBUTING.md:156]: (Everything above is just drive-by formatting changes added by me after the initial PR.)
